build_tools/cmake/iree_macros.cmake

```diff
@@ -60,10 +60,6 @@ function(iree_package_ns PACKAGE_NS)
   string(REPLACE ${IREE_ROOT_DIR} "" _RELATIVE_PATH ${CMAKE_CURRENT_LIST_DIR})
   string(SUBSTRING ${_RELATIVE_PATH} 1 -1 _RELATIVE_PATH)
 
-  # If changing the directory/package mapping rules, please also implement
-  # the corresponding rule in:
-  #   build_tools/bazel_to_cmake/bazel_to_cmake_targets.py
-  # Some sub-trees form their own roots for package purposes. Rewrite them.
   if(_RELATIVE_PATH MATCHES "^compiler/src/(.*)")
```

Comment:
This bit still seems really complicated. Seems like we should be able to simplify it a lot. Like why is there the special carveout for defs?

Reply:
Good point. We could also make the `defs` targets use the fallthrough. They're only used in one place each:

https://github.com/google/iree/blob/ec5840a6b84399506eaf871ac8c3ee03f5557769/compiler/src/CMakeLists.txt#L16-L19 https://github.com/google/iree/blob/ec5840a6b84399506eaf871ac8c3ee03f5557769/runtime/src/CMakeLists.txt#L16-L18

Reply:
Removed the special casing for `defs` and updated the places that referenced them